### ISSUE TITLE
Layout/ClassStructure bad autocorrection

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#6962]
I'm expecting RuboCop to always produce equivalent code when autocorrecting.

Layout/ClassStructure cop autocorrects badly

```ruby
class Foo
  def initialize; end

  local = 42
  EXAMPLE = local * 2
end

# gets autocorrected to:
class Foo
  EXAMPLE = local * 2
  def initialize; end

  local = 42
end
```

I believe that autocorrection should be disabled unless the constant is set to an independent literal (e.g. `CONSTANT = 42`). Even then, the way the code is currently moved around is wrong, like this HEREDOC case shows:

```ruby
class Foo
  def initialize; end

  TEXT = <<-EXAMPLE.freeze
  Hi!
  EXAMPLE
end

# autocorrected to:

class Foo
  TEXT = <<-EXAMPLE.freeze
  def initialize; end

  Hi!
  EXAMPLE
end
```

```
$ [bundle exec] rubocop -V
0.67.2 (using Parser 2.6.2.1, running on ruby 2.2.10 x86_64-darwin18)
```

**Comment by marcandre** [comment]
Bump

**Comment by allentsai** [comment]
We are seeing something similar.  Additionally:
```
Layout/ClassStructure:
  Enabled: true
  Categories:
    attribute:
      - attr_accessor
      - attr_writer
      - attr_reader
  ExpectedOrder:
    - module_inclusion
    - constants
    - association
    - public_attribute
    - public_delegate
    - macros
    - public_class_methods
    - initializer
    - public_methods
    - protected_methods
    - private_attribute
    - private_delegate
    - private_methods
```
has returned the following, post auto-correct:
```
    MY_CONSTANT = 10_000
    attr_reader :performed_at

    belongs_to :a_relation, class_name: 'ARelation::Base'
    has_many :sub_relations, class_name: 'SubRelation', foreign_key: 'sub_relation_id'
```
expected:
```
    MY_CONSTANT = 10_000

    belongs_to :a_relation, class_name: 'ARelation::Base'
    has_many :sub_relations, class_name: 'SubRelation', foreign_key: 'sub_relation_id'

    attr_reader :performed_at
```
Maybe I am misunderstanding what `association` means for the config...

**Comment by marcandre** [comment]
Bump
